custom_components/waste_collection_schedule/waste_collection_schedule/source/ccc_govt_nz.py

```diff
@@ -24,21 +50,37 @@ def fetch(self):
             "crs": "epsg:4326",
             "limit": 1,
         }
-        r = requests.get(
-            "https://opendata.ccc.govt.nz/CCCSearch/rest/address/suggest",
+
+        # Updated request using SSL code snippet
+        r = get_legacy_session().get("https://opendata.ccc.govt.nz/CCCSearch/rest/address/suggest",
             params=addressQuery,
+            # verify=False,
         )
+
+        # Original request code
+        # r = requests.get(
+        #     "https://opendata.ccc.govt.nz/CCCSearch/rest/address/suggest",
+        #     params=addressQuery,
+        # )
         address = r.json()
 
         # Find the Bin service by Rating Unit ID
         binsHeaders = {
             "client_id": "69f433c880c74c349b0128e9fa1b6a93",
             "client_secret": "139F3D2A83E34AdF98c80566f2eb7212"
         }
-        r = requests.get(
-            "https://ccc-data-citizen-api-v1-prod.au-s1.cloudhub.io/api/v1/properties/" + str(address[0]["RatingUnitID"]),
+
+        # Updated request using SSL code snippet
+        r = get_legacy_session().get("https://ccc-data-citizen-api-v1-prod.au-s1.cloudhub.io/api/v1/properties/" + str(address[0]["RatingUnitID"]),
```

Comment:
Could you create a single session and reuse it here?

Reply:
now only uses get_legacy_session